x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/legend_controls.tsx

```diff
@@ -181,184 +182,177 @@ export const LegendControls = ({
     : [];
 
   return (
-    <ControlContainer data-test-subj="legendControls">
-      <EuiPopover
-        isOpen={isPopoverOpen}
-        closePopover={handleCancelClick}
-        id="legendControls"
-        button={buttonComponent}
-      >
-        <EuiPopoverTitle>Legend Options</EuiPopoverTitle>
-        <EuiForm style={{ minWidth: 400 }}>
-          <EuiFormRow
-            display="columnCompressed"
-            label={i18n.translate('xpack.infra.legendControls.colorPaletteLabel', {
-              defaultMessage: 'Color palette',
-            })}
-          >
-            <>
-              <EuiSelect
-                options={PALETTE_OPTIONS}
-                value={draftLegend.palette}
-                id="palette"
-                onChange={handlePaletteChange}
-                compressed
-                data-test-subj="legendControlsPalette"
-              />
-              <EuiSpacer size="m" />
-              <PalettePreview
-                palette={draftLegend.palette}
-                steps={draftLegend.steps}
-                reverse={draftLegend.reverseColors}
-              />
-            </>
-          </EuiFormRow>
-          <EuiFormRow
-            display="columnCompressed"
-            label={i18n.translate('xpack.infra.legendControls.stepsLabel', {
-              defaultMessage: 'Number of colors',
-            })}
-          >
-            <EuiRange
-              id="steps"
-              min={2}
-              max={18}
-              step={1}
-              value={draftLegend.steps}
-              onChange={handleStepsChange}
-              showValue
-              fullWidth
+    <EuiPopover
+      isOpen={isPopoverOpen}
+      closePopover={handleCancelClick}
+      id="legendControls"
+      button={buttonComponent}
+      anchorPosition="leftCenter"
+      data-test-subj="legendControls"
+    >
+      <EuiPopoverTitle>Legend Options</EuiPopoverTitle>
+      <EuiForm style={{ minWidth: 400 }}>
```

Comment:
This minWidth doesn't lay out great on smaller mobile devices. I've opened https://github.com/elastic/kibana/issues/126999 to address that since this PR doesn't change anything compared to main.